### PR TITLE
fix NaN bug in orthographic projections and add tests

### PR DESCRIPTION
**Pull request by sashankaryal**
## What changes are proposed in this pull request?

- Fix bug where NaNs were causing orthographic projection computations to crash
- Add E2E tests for pcd only dataset + orthographic projections
- Allow different shape generations in pcd media factory + ability to create pcds with NaNs
- Upgrade playwright to 1.37.1
- Add `panSample` method to modal pom

## How is this patch tested? If it is not, please explain why.

Local E2E

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

- Fix bug where NaNs were causing orthographic projection computations to crash

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: e2e-pw/src/shared/media-factory/pcd.ts [hunk 1]
```diff
@@ -8,17 +9,37 @@ import { Duration, getPythonCommand } from "src/oss/utils";
...
   pcd = o3d.geometry.PointCloud()
-  pcd.points = o3d.utility.Vector3dVector([[i, i, i] for i in range(${numPoints})])
-  o3d.io.write_point_cloud('${outputPath}', pcd)`;
 
-  const pythonCodeSingleLine = pythonCode.split("\n").join(";");
-  const command = getPythonCommand(["-c", `"${pythonCodeSingleLine}"`]);
+  if "${options.shape}" == "diagonal":
+    points = [[i, i, i] for i in range(${numPoints})]
+  elif "${options.shape}" == "cube":
+    loop_stop = ${numPoints} // 3
```
Comment (sashankaryal) [inline e2e-pw/src/shared/media-factory/pcd.ts:28]: need a cube root of `numPoints` here, oops

**Review by imanjra — APPROVED** [review]
LGTM!

File: e2e-pw/src/shared/media-factory/pcd.ts [hunk 2]
```diff
@@ -8,17 +9,37 @@ import { Duration, getPythonCommand } from "src/oss/utils";
...
-  const pythonCode = `import open3d as o3d
+  const pythonCode = `
+  import open3d as o3d
   pcd = o3d.geometry.PointCloud()
-  pcd.points = o3d.utility.Vector3dVector([[i, i, i] for i in range(${numPoints})])
-  o3d.io.write_point_cloud('${outputPath}', pcd)`;
 
-  const pythonCodeSingleLine = pythonCode.split("\n").join(";");
-  const command = getPythonCommand(["-c", `"${pythonCodeSingleLine}"`]);
+  if "${options.shape}" == "diagonal":
```
Comment (benjaminpkane) [inline e2e-pw/src/shared/media-factory/pcd.ts:25]: Clever 🚀

**Review by benjaminpkane — APPROVED** [review]
Also LGTM 👍

File: e2e-pw/src/oss/specs/3d/pcd-only-dataset.spec.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+  test("orthographic projections are rendered correctly", async ({
+    modal,
+    grid,
+  }) => {
+    await expect(grid.firstFlashlightSection).toHaveScreenshot(
+      "orthographic-projection-grid-cuboids.png",
+      {
+        animations: "allow",
+        mask: [modal.locator.getByTestId("looker3d-action-bar")],
```
Comment (benjaminpkane) [inline e2e-pw/src/oss/specs/3d/pcd-only-dataset.spec.ts:70]: Perhaps there is a way to avoid the mask by ensuring Looker3D is blurred (out of focus)

**Pull request merged**